ExperimentTwitter/TweetsSearchViewController.m

```diff
@@ -0,0 +1,133 @@
+    _searchController.searchBar.placeholder = @"Search Twitter";
+    _searchController.searchBar.delegate = self;
+
+    self.navigationItem.rightBarButtonItem = nil;
+    self.navigationItem.leftBarButtonItem = nil;
+}
+
+- (void) searchBarTextDidBeginEditing:(UISearchBar *)searchBar {
+    if(searchBar == _searchController.searchBar) {
+        searchBar.placeholder = @"Search Twitter";
+    }
+}
+
+- (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar {
+    if(searchBar == _searchController.searchBar) {
+        NSLog(@"Search Button Pressed with query %@", _searchController.searchBar.text);
+        self.searchText = _searchController.searchBar.text;
+        _searchController.active = NO;
+    }
+}
+
+- (void) setSearchText:(NSString *)searchText {
+    [self fetchUsers:searchText];
+    [self fetchTweets:searchText];
+    [self.tableView reloadData];
+}
+
+-(void)dealloc {
+    [_searchController.view removeFromSuperview];
+}
+
+
+-(void) fetchTweets:(NSString *)searchText {
+    NSManagedObjectContext *managedObjectContext = [CoreDataHelper managedObjectContext];
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] initWithEntityName:@"Tweet"];
+    [fetchRequest setPredicate:[NSPredicate predicateWithFormat:@"text CONTAINS [cd] %@",searchText]];
+    NSSortDescriptor *createdAtSortDescriptor;
+    createdAtSortDescriptor = [[NSSortDescriptor alloc] initWithKey:@"createdAt"
+                                                          ascending:NO];
+    [fetchRequest setSortDescriptors:[NSArray arrayWithObjects:createdAtSortDescriptor, nil]];
+    self.tweets = [managedObjectContext executeFetchRequest:fetchRequest error:nil];
+    NSLog(@"Fetched %ld tweets for search query %@", [self.tweets count], searchText);
+}
+
+
+-(void) fetchUsers:(NSString *)searchText {
+    NSManagedObjectContext *managedObjectContext = [CoreDataHelper managedObjectContext];
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] initWithEntityName:@"User"];
+    [fetchRequest setPredicate:[NSPredicate predicateWithFormat:@"name CONTAINS [cd] %@",searchText]];
+    NSSortDescriptor *nameSortDescriptor = [[NSSortDescriptor alloc] initWithKey:@"name" ascending:YES];
+    [fetchRequest setSortDescriptors:[NSArray arrayWithObjects:nameSortDescriptor, nil]];
+    self.users = [managedObjectContext executeFetchRequest:fetchRequest error:nil];
+    NSLog(@"Fetched %ld users for search query %@", self.users.count, searchText);
+}
+
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 2 ;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return (section == 0) ? self.users.count : self.tweets.count;
+}
+
+- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
+    if([self tableView:tableView numberOfRowsInSection:section] == 0) {
+         return nil;
+    }
+    else {
```

Comment:
`else` should be in the same line as above.
Also `else` is not even required here.